Pegasus/include/particlecontacts.hpp

```diff
@@ -1,49 +1,46 @@
-#ifndef PEGAS_PARTICLE_CONTACTS_HPP
-#define PEGAS_PARTICLE_CONTACTS_HPP
+#ifndef PEGASUS_PARTICLE_CONTACTS_HPP
+#define PEGASUS_PARTICLE_CONTACTS_HPP
 
 #include "Pegasus/include/geometry.hpp"
 #include "Pegasus/include/mechanics.hpp"
-#include "Pegasus/include/particlecontacts.hpp"
 #include "Pegasus/include/particle.hpp"
+
 #include <vector>
 
 namespace pegasus {
 
 class ParticleContact {
 public:
-    using Ptr = std::shared_ptr<ParticleContact>;
-
-public:
-    ParticleContact(Particle::Ptr const& a, Particle::Ptr const& b,
-        real const restitution, Vector3 const& contactNormal,
-        real const penetration);
+    ParticleContact(Particle::Ptr const a, Particle::Ptr const b,
+        double const restitution, Vector3 const& contactNormal,
+        double const penetration);
 
-    void resolve(real const duration) const;
-    real calculateSeparatingVelocity() const;
+    void resolve(double const duration) const;
+    double calculateSeparatingVelocity() const;
 
 private:
     Particle::Ptr mA;
     Particle::Ptr mB;
-    real mRestitution;
-    Vector3 const mContactNormal;
-    real mPenetration;
+    double mRestitution;
+    Vector3 mContactNormal;
+    double mPenetration;
 
 private:
-    void resolveVelocity(real const duration) const;
+    void resolveVelocity(double const duration) const;
 
-    void resolveInterpenetration(real const duration) const;
+    void resolveInterpenetration(double const duration) const;
 };
 
-using ParticleContactsArray = std::vector<ParticleContact::Ptr>;
+using ParticleContacts = std::vector<ParticleContact>;
```

Comment:
Since it is a vector of actual elements and not their pointers, we're working with 56byte (if I'm not mistaken) objects. How many can there be? I am a bit concerned about vector growth